site/en/articles/urlpattern/index.md

```diff
@@ -0,0 +1,412 @@
+patterns.
+
+When using named groups in a pattern, the matches will exposed as properties
+whose names correspond to each group name.
+
+{% Aside %}
+If you're familiar with the
+<code>[exec()](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec)</code>
+method of a
+<code>[RegExp](https://developer.mozilla.org/docs/Web/JavaScript/Guide/Regular_Expressions)</code>
+object, then the return values from <code>URLPattern.exec()</code>
+method will look
+[very similar](https://developer.mozilla.org/docs/Web/JavaScript/Guide/Regular_Expressions/Groups_and_Ranges#examples)
+when it comes to representing anonymous and named groups.
+{% endAside %}
+
+### Unicode support and normalization
+
+`URLPattern` supports Unicode characters in a few different ways.
+
+- Named groups, like `:café`, can contain Unicode characters. The rules used for
+  valid
+  [JavaScript identifiers](https://developer.mozilla.org/docs/Glossary/Identifier)
+  apply to named groups.
+
+- Text within a pattern will be automatically encoded according to the same
+  rules used for URL encoding of that particular component. Unicode characters
+  within `pathname` will be
+  [percent-encoded](https://developer.mozilla.org/docs/Glossary/percent-encoding),
+  so a `pathname` pattern like `/café` is normalized to `/caf%C3%A9`
+  automatically. Unicode characters in the `hostname` are automatically encoded
+  using [Punycode](https://en.wikipedia.org/wiki/Punycode), rather than
+  percent-encoding.
+
+- Regular expression groups must contain only ASCII characters. Regular
+  expression syntax makes it difficult and unsafe to automatically encode
+  Unicode characters in these groups. If you want to match a Unicode character
+  in a regular expression group, you need to percent encode it manually, like
+  `(caf%C3%A9)` to match `café`.
+
+In addition to encoding Unicode characters, `URLPattern` also performs URL
+normalization. For example, `/foo/./bar` in the `pathname` component is
+collapsed to the equivalent `/foo/bar`.
+
+When in doubt about how a given input pattern has been normalized, inspect the
+constructed `URLPattern` instance using your browser's
+[DevTools](https://developer.mozilla.org/docs/Learn/Common_questions/What_are_browser_developer_tools).
+
+## Putting it all together
+
+The Glitch demo embedded below illustrates a core use case of `URLPattern`
+inside of a service worker's
+<code>[fetch event handler](https://developer.mozilla.org/docs/Web/API/Service_Worker_API/Using_Service_Workers#custom_responses_to_requests)</code>,
+mapping specific patterns to asynchronous functions that could generate a
+response to network requests. The concepts in this example could be applied to
+other routing scenarios as well, either server-side or client-side.
+
+{% Glitch {
+id: 'urlpattern-sw-routing',
+path: 'service-worker.js',
+previewSize: 35
+} %}
+
+## Browser support and polyfills
```

Comment:
Can we move this support bit to the top somewhere, just under the initial description.